ide/static/ide/js/editor.js

```diff
@@ -252,80 +272,101 @@ CloudPebble.Editor = (function() {
+                                simply: true,
+                                util2: true,
+                                ajax: true
+                            });
+                        } else if (CloudPebble.ProjectInfo.type == 'pebblejs') {
+                            _.extend(jshint_globals, {
+                                require: true,
+                                ajax: true
+                            });
+                        } else if (CloudPebble.ProjectInfo.app_modern_multi_js) {
+                            _.extend(jshint_globals, {
+                                require: true,
+                                exports: true,
+                                module: true
+                            });
+                        }
 
 
-                    var success = JSHINT(code_mirror.getValue(), {
-                        freeze: true,
-                        evil: false,
-                        immed: true,
-                        latedef: "nofunc",
-                        undef: true,
-                        unused: "vars"
-                    }, jshint_globals);
-                    if(!success) {
-                        _.each(JSHINT.errors, function(error) {
-                            // It is apparently possible to get null errors; omit them.
-                            if(!error) return;
-                            // If there are multiple errors on one line, we'll have already placed a marker here.
-                            // Instead of replacing it with a new one, just update it.
-                            var markers = code_mirror.lineInfo(error.line - 1).gutterMarkers;
-                            if(markers && markers['gutter-hint-warnings']) {
-                                var marker = $(markers['gutter-hint-warnings']);
-                                marker.attr('title', marker.attr('title') + "\n" + error.reason);
-                            } else {
-                                var warning = $('<i class="icon-warning-sign icon-white"></span>');
-                                warning.attr('title', error.reason);
-                                code_mirror.setGutterMarker(error.line - 1, 'gutter-hint-warnings', warning[0]);
-                                warning_lines.push(code_mirror.addLineClass(error.line - 1, 'background', 'line-hint-warning'));
-                            }
-                        });
+                        var success = JSHINT(code_mirror.getValue(), {
+                            freeze: true,
+                            evil: false,
+                            immed: true,
+                            latedef: "nofunc",
+                            undef: true,
+                            unused: "vars"
+                        }, jshint_globals);
+                        if (!success) {
+                            errors = JSHINT.errors;
+                        }
+                    }
+                    else {
```

Comment:
Oh, that was an oversight, although it looks like it appears in a few places. I could fix it in this PR although I'd be tempted to wait until all the PRs are done and then fix all the formatting in one fell swoop.